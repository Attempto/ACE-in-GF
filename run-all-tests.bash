#!/bin/bash

# @author Kaarel Kaljurand
# @version 2012-04-04

parser="Parser"

pgf="ACE-0_0_2.pgf"
#lang="TestAttemptoAce"
lang="ClexAce"
tests=tests

test_out="test_out.txt"
test_out_fail="test_out_fail.txt"

for dir in $(find $tests -maxdepth 1 -mindepth 1 -type d)
do
	echo "Testing: $dir"
	in="${dir}/sentences.txt"
	out="${dir}/${test_out}"

	time ./${parser} $pgf $lang < $in > $out

	echo -n "ok: "
	cat $out | grep "|OK" | wc -l

	echo -n "FAIL: "
	cat $out | grep "|FAIL" | wc -l

	cat $out | grep "|FAIL" | sed "s/^.*|FAIL //" | sort | uniq -c | sort -nr > ${dir}/${test_out_fail}
	echo
done
