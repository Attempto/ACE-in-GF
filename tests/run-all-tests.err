Parser: <stdin>: hGetLine: end of file

real	3m5.535s
user	3m4.424s
sys	0m0.512s
Parser: <stdin>: hGetLine: end of file

real	0m0.153s
user	0m0.120s
sys	0m0.016s
Parser: <stdin>: hGetLine: end of file

real	0m1.580s
user	0m1.520s
sys	0m0.028s
Parser: <stdin>: hGetLine: end of file

real	0m0.315s
user	0m0.280s
sys	0m0.008s
Parser: <stdin>: hGetLine: end of file

real	0m22.649s
user	0m22.429s
sys	0m0.104s
