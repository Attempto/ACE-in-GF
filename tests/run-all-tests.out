Testing: tests/acewiki_aceowl
ok: 17170
FAIL: 2252

Testing: tests/no
ok: 8
FAIL: 0

Testing: tests/geo
ok: 9
FAIL: 826

Testing: tests/ontograph_40
ok: 6
FAIL: 34

Testing: tests/ace
ok: 1454
FAIL: 1263

